uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -411,9 +412,15 @@ private static function validateLeaveDatesDoesNotOverlapContracts($params) {
       'end_date' => $toDate->format('Y-m-d'),
     ]);
 
-    if ($contract['count'] > 1) {
+    $contractForEndDate = civicrm_api3('HRJobContract', 'getcontractswithdetailsinperiod', [
```

Comment:
If it is always singular then why check count?

It's not that clear what's happening when looking at this. It looks to me like it's checking for a contract that has a start and end date that are both on the final day of the leave request.

The exception message sounds like it's saying the contact has no contracts that contain the leave request period. 

Shouldn't the validation check if the contact has any contract that starts _before_ the leave request start and ends _after_ the leave request ends? I might be misinterpreting this, but it's not so clear when I look at it what's happening.